Objects/longobject.c

```diff
@@ -1965,7 +1965,9 @@ long_to_decimal_string_internal(PyObject *aa,
     digit *pout, *pin, rem, tenpow;
     int negative;
     int d;
-    int kind;
+    int kind = 0;
+
+    assert(writer == NULL || bytes_writer == NULL);
```

Comment:
What's the purpose of this assertion? Make sure that both are not set at the same time?

Reply:
yes

Reply:
I would prefer to write it more explicitly:

```
// writer or bytes_writer can be used, but not both at the same time.
assert(!(writer != NULL && bytes_writer != NULL));
```